### ISSUE TITLE
Allow properties to throw NotImplementedException

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn-analyzers#1819]
Analyzer package: [Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)
Package Version: 2.6.2-beta2
Diagnostic ID: CA1065

This kind of thing is common in test assemblies:

```c#
public sealed class TestFoo : Foo
{
    public void Bar()
    {
        // Implementation for test purposes
    }

    public int Baz => throw new NotImplementedException();
}
```

> CA1065 get_Baz  creates an exception of type NotImplementedException, an exception type that should not be raised in a property. If this exception instance might be raised, use a different exception type, convert this property into a method, or change this property's logic so that it no longer raises an exception.

In test code, NotImplementedException is exactly what should happen if the property ever ends up being called.

**Comment by mavasani** [comment]
@jnm2 The analyzers do not have any specialization for test or product code. It is the end user's responsibility to disable product-only rules for test projects.
